mono/utils/mono-proclib.c

```diff
@@ -1017,16 +1019,19 @@ mono_pe_file_map (gunichar2 *filename, gint32 *map_size, void **handle)
 	int fd = -1;
 	struct stat statbuf;
 	gpointer file_map = NULL;
+	ERROR_DECL (err);
 
 	/* According to the MSDN docs, a search path is applied to
 	 * filename.  FIXME: implement this, for now just pass it
 	 * straight to open
 	 */
 
-	filename_ext = mono_unicode_to_external (filename);
+	filename_ext = mono_unicode_to_external_checked (filename, err);
+	// Assert added to diagnose https://github.com/mono/mono/issues/14730, remove after resolved
+	g_assertf (filename_ext != NULL, "%s: unicode conversion returned NULL; %s; 0x%hx", __func__, mono_error_get_message (err), filename);
```

Comment:
I don't think this is quite what we want.   if `filename_ext` is null, this will print the conversion error message and just the first `short` from the address of `filename`.
I think we want something more like
```c
if (G_UNLIKELY (filename_ext == NULL)) {
    GString *raw_bytes = g_string_new (NULL);
    char *p = (char*)filename;
    while (*p != 0) {
        g_string_append_printf (raw_bytes, " %02X", (int)*p);
        p++;
    }
    g_assertf (filename_ext != NULL, "%s: unicode conversion returned NULL; %s; input was: %s", __func__, mono_error_get_message (err), raw_bytes->str);
    g_string_free (raw_bytes, TRUE);
}
```

Reply:
Yeah, whoops. Not sure what I was thinking there, but it's been updated.